CRM/Export/Form/Map.php

```diff
@@ -146,6 +146,14 @@ protected function getPreviewData() {
       $exportParams['postal_mailing_export']['postal_mailing_export'] == 1
     );
     $processor = new CRM_Export_BAO_ExportProcessor($this->get('exportMode'), NULL, $this->get('queryOperator'), $this->get('mergeSameHousehold'), $isPostalOnly, $this->get('mergeSameAddress'));
+    // dev/core#1775 Remove notes,groups and tags from the preview data to speed up export
+    $defaultProperties = $processor->getDefaultReturnProperties();
+    foreach ($defaultProperties as $key => $var) {
+      if (in_array($key, ['groups', 'tags', 'notes'])) {
+        unset($defaultProperties[$key]);
+      }
+    }
```

Comment:
I don't understand the purpose of this foreach loop. You are just unsetting 3 keys from an array. FYI `unset()` does not raise any error if the key isn't there. I think this could be simplified to:

```suggestion
    CRM_Utils_Array::remove($defaultProperties, 'groups', 'tags', 'notes');
```